### PR TITLE
Vault Integration for orakl api

### PR DESCRIPTION
**Pull request by jo-bisonai**
# Description

This PR introduces integration of Vault Secrets via Kubernetes pod access. Testing is restricted to Kubernetes pods, as only service accounts within these pods can successfully load secrets into the application.

After thorough testing and successful deployment for testing purposes, the integration has proven to function effectively.

For the migration period in our CI/CD pipeline, the Helm chart section has been temporarily commented out.

### CODE REVIEW

File: api/secrets/secrets.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"context"
+	"fmt"
+
+	vault "github.com/hashicorp/vault/api"
+	auth "github.com/hashicorp/vault/api/auth/kubernetes"
+)
+
+type SecretEnv struct {
+	VaultRole       string
+	JwtPtah         string
```
[the remark this answers is not in the document]
Comment (jo-bisonai) [reply api/secrets/secrets.go]: fixed!

File: api/utils/utils.go [hunk 2]
```diff
@@ -264,12 +265,37 @@ func DecryptText(encryptedText string) (string, error) {
...
+	vaultRole := os.Getenv("VAULT_ROLE")
+	jwtPath := os.Getenv("JWT_PATH")
+	vaultSecretPath := os.Getenv("VAULT_SECRET_PATH")
+	vaultKeyName := os.Getenv("VAULT_KEY_NAME")
+
+	if vaultRole != "" && jwtPath != "" && vaultSecretPath != "" && vaultKeyName != "" {
+		log.Println("Using Vault to get secrets")
+		secretsEnv := secrets.SecretEnv{
+			VaultRole:       vaultRole,
+			JwtPtah:         jwtPath,
```
[the remark this answers is not in the document]
Comment (jo-bisonai) [reply api/utils/utils.go]: fixed

File: api/secrets/secrets.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	k8sAuth, err := auth.NewKubernetesAuth(
+		s.VaultRole,
+		auth.WithServiceAccountTokenPath(s.JwtPtah),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize Kubernetes auth method: %w", err)
+	}
+
+	authInfo, err := client.Auth().Login(context.TODO(), k8sAuth)
```
[the remark this answers is not in the document]
Comment (jo-bisonai) [reply api/secrets/secrets.go]: What a great review! Thanks

**Review by nick-bisonai — APPROVED** [review]
LGTM!

**Pull request merged**